File: modules/get_timezone.py
import sys
from timezonefinder import TimezoneFinder
import pytz
from datetime import datetime
import json

if __name__ == "__main__":
    if len(sys.argv) < 3:
        print(json.dumps({"error": "Usage: python get_timezone.py <latitude> <longitude>"}))
        sys.exit(1)

    try:
        latitude = float(sys.argv[1])
        longitude = float(sys.argv[2])
    except ValueError:
        print(json.dumps({"error": "Invalid latitude or longitude."}))
        sys.exit(1)

    tf = TimezoneFinder()
    timezone_str = tf.timezone_at(lng=longitude, lat=latitude)

    if timezone_str:
        try:
            tz = pytz.timezone(timezone_str)
            now = datetime.now(tz)
            current_hour = now.strftime("%H")
            current_minute = now.strftime("%M")

            print(json.dumps({
                "timezone": timezone_str,
                "hour": current_hour,
                "minute": current_minute
            }))
        except pytz.UnknownTimeZoneError:
            print(json.dumps({"error": f"Unknown timezone: {timezone_str}"}))
            sys.exit(1)
    else:
        print(json.dumps({"error": "Unknown timezone."}))
        sys.exit(1)

File: modules/get_timezone.php
<?php
header('Content-Type: application/json');

$latitude = $_POST['latitude'] ?? null;
$longitude = $_POST['longitude'] ?? null;

if (!$latitude || !$longitude) {
    echo json_encode(['error' => 'Missing latitude or longitude']);
    exit;
}

$pythonServiceUrl = "http://127.0.0.1:5000/get_timezone?latitude=" . $latitude . "&longitude=" . $longitude;

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $pythonServiceUrl);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
$response = curl_exec($ch);
$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

if ($response === false) {
    echo json_encode(['error' => 'Failed to connect to Python service.']);
    exit;
}

$jsonData = json_decode($response, true);

if ($jsonData === null) {
    echo json_encode(['error' => 'Invalid JSON response from Python service', 'raw_output' => $response]);
    exit;
}

if ($httpCode !== 200) {
    echo json_encode(['error' => 'Python service error', 'details' => $jsonData['error'] ?? 'Unknown error']);
    exit;
}

echo json_encode($jsonData);
